### DISCUSSION TITLE
Contribute to SB Lern

### DISCUSSION THREAD

**Discussion by rhwarrier** [Sunbird-Lern/Community#15] [Contribution]
Successful contributions to Sunbird can be in any form. For ex., one can contribute by: 
> Contributing by actively engaging with the community on the discussion forum
> Contributing by identifying the needs i.e. bugs, documentation and installation issue, proposing new features and proposing enhancements to existing features
> Contributing by developing new features, enhancements or fixing bugs
> Contributing documentation changes
> Contributing installation related changes
> Contributing towards release testing

Everyone will have access to create a new discussion, add comments to existing discussions and tag groups or individuals to the discussion thread. 

**How to start ?**

Every Building block publishes their list of [issues](https://project-sunbird.atlassian.net/jira/software/c/projects/LR/issues/?jql=project%20%3D%20%22LR%22%20AND%20type%20in%20standardIssueTypes%28%29%20ORDER%20BY%20created%20DESC) with the latest status in the issue tracker and the link for the same is available on the microsite as well as discussion forum landing page. The issues that are not yet taken up for development are published as a backlog of issues on the issue tracker and the link for the same is available on the microsite. The issues in the backlog are available to be taken up for development

Every issue will have a complexity field which will give a sense of ease or difficulty involved in developing the issues. The issues will also have a Priority field which will give a sense of urgency of developing the issue. Both these fields will help in identifying the appropriate issue to be picked up for development.

Once an issue is picked up for development, updating the status of the issue by the contributor in the issue tracker is very important as it keeps the community updated on the latest status of the issue. It also helps to plan the testing and other release related activities. 

Once the development is completed by the contributor, the changes are finally validated and tested by the community QA team to sign-off the changes for inclusion in the latest version of the building block. Any issues found during this testing has to be fixed by the contributor and validated by the QA team within the published release timelines

Additional details on the Contribution Workflows can be found on the [Sunbird Community Practices](https://sunbird.gitbook.io/sunbird-community-practices/VJZlxBFjATn5Y97QSiJB/) document.

**Comment by Rudrant** [comment]
I have setup the sunbird portal on my local machine. Where can I get the configuration details for testing purpose? Did your team setup the backend services for testing? If anybody is willing to contribute to this project how about the APIs?

**Comment by rhwarrier** [comment]
The Sunbird ED discussion forum may be able to guide you better on how to take this forward. (https://github.com/Sunbird-Ed/Community/discussions)
Tagging @luckynvdu and @rajeevsathish for further inputs.
Relevant documentation regarding the APIs can be found in the respective microsites - nay specific ones you're looking for further information on ?
